src/main/java/teammates/sqllogic/core/FeedbackResponsesLogic.java

```diff
@@ -176,6 +177,56 @@ private List<FeedbackResponse> getFeedbackResponsesFromTeamForQuestion(
         return responses;
     }
 
+    /**
+     * Updates a non-null feedback response by {@link FeedbackResponse}.
+     *
+     * <p>Cascade updates its associated feedback response comment
+     * (e.g. associated response ID, giverSection and recipientSection).
+     *
+     * <p>If the giver/recipient field is changed, the response is updated by recreating the response
+     * as question-giver-recipient is the primary key.
+     *
+     * @return updated feedback response
+     * @throws InvalidParametersException if attributes to update are not valid
+     * @throws EntityDoesNotExistException if the comment cannot be found
+     */
+    public FeedbackResponse updateFeedbackResponseCascade(FeedbackResponse feedbackResponse)
+            throws InvalidParametersException, EntityDoesNotExistException {
+
+        FeedbackResponse oldResponse = frDb.getFeedbackResponse(feedbackResponse.getId());
+        FeedbackResponse newResponse = frDb.updateFeedbackResponse(feedbackResponse);
```

Comment:
I feel like this line causes the previous test to fail. Perhaps you can compare feedbackResponse with oldResponse instead before calling update? I think updateFeedbackResponse causes the merge to save the comments that have changed in your test

I tried referring to this but didn't 100% understand it
https://docs.jboss.org/hibernate/orm/6.1/userguide/html_single/Hibernate_User_Guide.html#pc-merge-gotchas

Reply:
In the failing test, the function only calls this line, but does not call `               frcLogic.updateFeedbackResponseComment(oldResponseComment);`. This is unintended behavior, as we did not specify the comments to cascade (https://docs.jboss.org/hibernate/orm/6.1/userguide/html_single/Hibernate_User_Guide.html#pc-cascade). 

Right now, I'm still trying to find out why the cascading automatically happens even when we did not state for the cascade to happen. Furthermore, it's an issue as I don't think we're planning for automatic cascade to be intended behavior in the future for performance reasons.

Reply:
Ah I see, didn't realise you removed the `Persist` from the previous PR review. Should we leave put this on hold till we figure this out?

Reply:
I was hoping to investigate this and get it merged with the original OPTIMIZED PR too, but we're rushing to merge `SubmitFeedbackResponseAction`'s migration as it's a large part of other E2E tests. These 2 issues do not contribute much to the behavior of the program, but more on the performance side, so we'll leave that to later to improve on.